src/main/java/org/cirdles/topsoil/dataset/DatasetResourceFactory.java

```diff
@@ -15,22 +15,22 @@
  */
 package org.cirdles.topsoil.dataset;
 
-import java.util.List;
+import java.nio.file.Path;
 
 /**
  *
- * @author John Zeringue
+ * @author parizotclement
  */
-public interface DatasetManager {
+public class DatasetResourceFactory {
 
-    public List<Dataset> getDatasets();
+    private final Path basePath;
 
-    public boolean isOpen(Dataset dataset);
+    public DatasetResourceFactory(Path basePath) {
+        this.basePath = basePath;
```

Comment:
`DatasetReader`s and `DatasetWriter`s should be configurable from here as well, although you should be selective about what's able to be set from the constructor.